### DISCUSSION TITLE
Branch spin-out questions

### DISCUSSION THREAD

**Discussion by Amleto** [NeogitOrg/neogit#1358] [General]
What is the use case for branch spin-out? Isn't it just branch create at head? 

I also noticed:

1. I don't see the "staying on HEAD due to uncommitted changes" notification
2. spin_off_branch function does not fire events like the create_branch function does 

Are either of those bugs that should be filed?

**Comment by CKolkey** [comment]
Good question - Spin off/out are particularly useful if your branch has an upstream configured. If, for example, you make some commits to your local branch and before pushing them to the upstream decide they are better off on their own branch, spin off/out will take those commits, stick them into a new branch, and reset the current branch to it's upstream. You can read more in Magit's documentation: https://magit.vc/manual/magit/Branch-Commands.html

To your other points, you should only see that message if you select "spin out", not "spin off", with uncommitted changes in your index. Basically it means that it's going to act like "spin off" instead, and update your HEAD. If it's _still_ not showing a notice, then that's a bug, yeah.

And not firing an event is just an oversight on my part. Easy fix.

**Reply by Amleto** [reply]
Oh interesting! Thanks for the magit link.

Yeah, I thought my neogit maps were broken because I was pressing `S` for spin-out, but getting no notification and "spin off" was showing in the prompt (I had uncommitted changes)

How should the notification be visible?

**Reply by CKolkey** [reply]
It just uses `vim.notify()` under the hood, so however you have that set up :)

**Reply by Amleto** [reply]
I don't have anything myself. I use fidget so I presume that should be linked somehow.

If I disable that, what do default notify() messages look like? Where to find them?

**Reply by CKolkey** [reply]
Hmm. I think they just show up in the row above the cmd bar at the bottom? You can test it with `:lua vim.notify("hello")`

**Reply by Amleto** [reply]
Hmm. fidget doesn't override by default. I can see the message when doing the:lua call like you say.
Whether I configure fidget to override vim.notify or when I'm using default behaviour I do not see the notification that should inform of "staying on HEAD..."

**Comment by alsmnn** [comment]
The link for the documentation has changed, just fyi
https://docs.magit.vc/magit/Branch-Commands.html
